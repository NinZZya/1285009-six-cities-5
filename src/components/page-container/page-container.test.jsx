import React from 'react';
import renderer from 'react-test-renderer';
import PageContainer from './page-container';


const PAGE_CONTAINER_TYPE = `index`;

describe(`Should PageContainer render correctly`, () => {
  it(`Should PageContainer render correctly without type and children`, () => {
    const tree = renderer
      .create(
          <PageContainer>
            <div>Div 1</div>
            <div>Div 2</div>
          </PageContainer>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should PageContainer render correctly with type and children`, () => {
    const tree = renderer
      .create(
          <PageContainer type={PAGE_CONTAINER_TYPE}>
            <div>Div 1</div>
            <div>Div 2</div>
          </PageContainer>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`Should PageContainer render correctly with type, empty and children`, () => {
    const tree = renderer
      .create(
          <PageContainer empty type={PAGE_CONTAINER_TYPE}>
            <div>Div 1</div>
            <div>Div 2</div>
          </PageContainer>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
