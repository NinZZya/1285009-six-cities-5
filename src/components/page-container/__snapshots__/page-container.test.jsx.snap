// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PageContainer render correctly Should PageContainer render correctly with type and children 1`] = `
<main
  className="page__main page__main--index "
>
  <div>
    Div 1
  </div>
  <div>
    Div 2
  </div>
</main>
`;

exports[`Should PageContainer render correctly Should PageContainer render correctly with type, empty and children 1`] = `
<main
  className="page__main page__main--index page__main--index-empty"
>
  <div>
    Div 1
  </div>
  <div>
    Div 2
  </div>
</main>
`;

exports[`Should PageContainer render correctly Should PageContainer render correctly without type and children 1`] = `
<main
  className="page__main page__main--undefined "
>
  <div>
    Div 1
  </div>
  <div>
    Div 2
  </div>
</main>
`;
